test/e2e/baseFixture.ts

```diff
@@ -0,0 +1,12 @@
+import { test as base } from "@playwright/test"
+import { CodeServer } from "./models/CodeServer"
+
+export const test = base.extend<{ codeServerPage: CodeServer }>({
+  codeServerPage: async ({ page }, use) => {
```

Comment:
This is a custom test-fixture. This means for every test a new one gets created. Since its using the page, which uses a context, Playwright does automatically take care of closing the context.

Reply:
Woah, that is super cool! That's fantastic! 🎉